loopy/transform/iname.py

```diff
@@ -1737,6 +1736,49 @@ def add_inames_to_insn(kernel, inames, insn_match):
 # }}}
 
 
+# {{{ remove_inames_from_insn
+
+@for_each_kernel
+def remove_inames_from_insn(kernel, inames, insn_match):
+    """
+    :arg inames: a frozenset of inames that will be added to the
+        instructions matched by *insn_match*, or a comma-separated
+        string that parses to such a tuple.
+    :arg insn_match: An instruction match as understood by
+        :func:`loopy.match.parse_match`.
+
+    :returns: a :class:`LoopKernel` with the *inames* removed from
+        the instructions matched by *insn_match*.
+
+    .. versionadded:: 2023.0
+    """
+
+    if isinstance(inames, str):
+        inames = frozenset(s.strip() for s in inames.split(","))
+
+    if not isinstance(inames, frozenset):
+        raise TypeError("'inames' must be a frozenset")
+
+    from loopy.match import parse_match
+    match = parse_match(insn_match)
+
+    new_instructions = []
+
+    for insn in kernel.instructions:
+        if match(kernel, insn):
+            new_inames = set(insn.within_inames)
+            for iname in inames:
+                new_inames.discard(iname)
```

Comment:
I wonder whether we should error if an iname is not part of the insn.

Reply:
Done in 5f87921b21e7e4fdc18b5633356643991178f9b0